vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -335,3 +335,22 @@ def _get_options(
         # changes. See https://pandas.pydata.org/docs/whatsnew/v2.1.0.html#whatsnew-210-enhancements-new-stack.
         targeted_data = pd.concat([targeted_data, pd.Series(current_value)]).stack().dropna()  # noqa: PD013
         return sorted(set(targeted_data) - {ALL_OPTION})
+
+    def _get_proposed_targets(self):
+        """Get all valid figure model targets for this control based on its location in the page hierarchy."""
+        page = model_manager._get_model_page(self)
+        page_containers = model_manager._get_models(model_type=Container, root_model=page)
+
+        # Find the control's parent model. Set it as the control's parent container it exists.
+        # Otherwise set it as the control's page.
+        root_model = next(
+            (
+                container
+                for container in page_containers
+                if any(control.id == self.id for control in container.controls)
+            ),
+            page,
+        )
+        return [
+            model.id for model in cast(Iterable[VizroBaseModel], model_manager._get_models(FIGURE_MODELS, root_model))
```

Comment:
```suggestion
            model.id for model in cast(Iterable[FigureType], model_manager._get_models(FIGURE_MODELS, root_model))
```

Assuming mypy doesn't object.